sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResource.java

```diff
@@ -0,0 +1,193 @@
+      @PathParam("collection") String collection,
+      @NotNull(message = "json schema not provided")
+          @RequestBody(
+              required = true,
+              description = "The JSON schema to attach",
+              content = {
+                @Content(
+                    mediaType = "application/json",
+                    examples =
+                        @ExampleObject(
+                            ref =
+                                "{\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}"))
+              })
+          JsonNode body) {
+    return jsonSchemaManager
+        .attachJsonSchema(
+            namespace, tableManager.getValidCollectionTable(namespace, collection), body)
+        .map(schema -> RestResponse.ok(new JsonSchemaDto(schema)));
+  }
+
+  @Operation(description = "Get a JSON schema from a collection.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace of the collection"),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to get the JSON schema from.")
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "200",
+            description = "Fetch successful.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = JsonSchemaDto.class,
+                          properties = @SchemaProperty(name = "schema", type = SchemaType.OBJECT)))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST),
+                      @ExampleObject(ref = OpenApiConstants.Examples.COLLECTION_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @GET
+  public Uni<RestResponse<Object>> getJsonSchema(
+      @Context UriInfo uriInfo,
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection) {
+
+    // go get the existing table
+    return jsonSchemaManager
+        .getJsonSchema(tableManager.getValidCollectionTable(namespace, collection))
+        .map(schema -> RestResponse.ok(new JsonSchemaDto(schema.requiredAt("/schema"))));
```

Comment:
and now here simply have that switch:

```
.onItem()
.IfNull()
.switch(() -> Uni.createFrom().failure(new ErrorCodeRuntimeException(ErrorCode.DOCS_API_JSON_SCHEMA_DOES_NOT_EXIST))
```

the nulls will be propagated along the chain, so there was no need to change the `getJsonSchema` and throw the exception there..